### ISSUE TITLE
Question : Need tutorials for creating widgets

### ISSUE THREAD

**Issue by fvoverscan** [algolia/instantsearch.js#3262]
Hi,

Firstly, I'm not an jquery expert.

I don't figure out how to custom a widget (RefinementList) to implement a multiple attributes name query.
For example, i need to query on best_offer.merchant.name or other_offers.merchant.name...

The [doc](https://community.algolia.com/instantsearch.js/v2/connectors/connectRefinementList.html#struct-CustomRefinementListWidgetOptions) only related to design the render... Or i miss something ?

The page related to custom widget is completly unclear (i read also all links on this page) : 
https://community.algolia.com/instantsearch.js/v2/guides/custom-widget.html

I don't understand what i have to do firstly.

Thanks for your Help

**Comment by Haroenv** [comment]
Hey @fvoverscan, what exactly do you mean with "implement a multiple attributes name query". Can you give a sketch of the UI you would like to achieve?

**Comment by fvoverscan** [comment]
Hi @Haroenv ,

Our datas structure :
hit : {
   best_offer : {
       merchant : {
           name: "AAA"
        }
   },
   other_offers :  [
     {
      merchant : {
         name: "BBB"
      }
    },
   ]
}

I would like to use a refinementList to ount all the product for merchants
For example : 
- AAA (10)
- BBB (5)
But i could use only one attribute "best_offer.merchant.name"  on refinementList.
I need both "best_offer.merchant.name" and "other_offers.merchant.name" and my structure can't be modified...
So i would like to create a custom widget but informations are a bit incomplet.

Thanks

**Comment by Haroenv** [comment]
This is a very advanced use case, but the way to go is: 

1. make a custom connector

You will have to base it on https://github.com/algolia/instantsearch.js/blob/develop/src/connectors/refinement-list/connectRefinementList.js

the main difference however should be that it doesn't need to work with 1 attribute, but with multiple. 

You will also need to do the sorting or merging of the lists yourself.

2. ignore the jQuery parts of the tutorial and use vanilla JS for the rendering

Basically on every "render" (every time the render function gets called in your connector) need to construct DOM for a refinement list to work as expected. 

Does that help you?

**Comment by fvoverscan** [comment]
Thanks for the reply.

Sorry, i don't understand where to add another attribute on the connector. I use the example on connectRefinementList but could you highligth where to add this ?

I updated my previous comments to explain more the use case.

Thanks

**Comment by fvoverscan** [comment]
Sorry i forgot to mention that i also need to facet search the query...

When i check/clic on merchant name, i would like to query on best_offer or other_offers merchant name.

Thanks

**Comment by Haroenv** [comment]
unfortunately, since this is _very_ custom code, I don't think we will be able to help you any further like this, you will need to modify the code which gets the refinements, to also get it from more than simply "attribute".
